src/cryptography/hazmat/_oid.py

```diff
@@ -46,21 +48,21 @@ def __init__(self, dotted_string: str):
                 % (self._dotted_string)
             )
 
-    def __eq__(self, other):
+    def __eq__(self, other: typing.Any) -> bool:
```

Comment:
If I don't annotate with `Any`, mypy will complain, yet the exception below still seems odd. I don't know of any other class where comparing to an instance of a _different_ class raises an exception. Why not just return `False` (same in many other places)?